src/tests/invoices.rs

```diff
@@ -1,24 +1,34 @@
+use std::sync::Once;
+
 use crate::api::app;
 use crate::api::invoices::{CreateInvoice, CreateInvoiceRow, PopulatedInvoice};
-use crate::models::NewParty;
+use crate::models::NewAddress;
 
 use axum::http::StatusCode;
+use axum::Router;
 use axum_test::multipart::MultipartForm;
 use axum_test::TestServer;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::EnvFilter;
 
+static INIT: Once = Once::new();
+async fn test_init() -> Router {
+    INIT.call_once(|| {
```

Comment:
I recommend we don't use Once, because if we do, tests may collide

Reply:
For more info see how crate::database::new() is implemented

Reply:
This `Once` is only for initing the `tracing_subscriber` so we get log output for the tests. trying to initialize `tracing_subscriber` multiple times results in a panic. Either it's `Once` or somehow check if `tracing_subscriber` has been initialized already?
But yeah, at least a comment would be good here.

Reply:
Oh yeah, didn't notice `crate::database::new()` was outside of `Once` :+1: